test/integration-test.sh

```diff
@@ -2,35 +2,76 @@
 
 # prevent souring of this script, only allow execution
 $(return >/dev/null 2>&1)
-test "$?" -eq "0" && { echo "This script should only be executed." >&2; exit 1;
-}
+test "$?" -eq "0" && { echo "This script should only be executed." >&2; exit 1; }
 
 # exit on errors, undefined variables, ensure errors in pipes are not hidden
-set -euo pipefail
+set -Eeuo pipefail
+
+# set log id and use shared log function for readable logs
+declare mydir
+mydir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+declare HOPR_LOG_ID="e2e-test"
+source "${mydir}/../scripts/utils.sh"
 
 usage() {
-  echo >&2
-  echo "Usage: $0 <node_api_1> <node_api_2> <node_api_3>" >&2
-  echo
-  echo >&2
+  msg
+  msg "Usage: $0 <node_api_1> <node_api_2> <node_api_3>"
+  msg
 }
 
 # return early with help info when requested
 ([ "${1:-}" = "-h" ] || [ "${1:-}" = "--help" ]) && { usage; exit 0; }
 
 # verify and set parameters
-test -z "${1:-}" && { echo "Missing first parameter" >&2; usage; exit 1; }
-test -z "${2:-}" && { echo "Missing second parameter" >&2; usage; exit 1; }
-test -z "${3:-}" && { echo "Missing third parameter" >&2; usage; exit 1; }
+test -z "${1:-}" && { msg "Missing first parameter"; usage; exit 1; }
+test -z "${2:-}" && { msg "Missing second parameter"; usage; exit 1; }
+test -z "${3:-}" && { msg "Missing third parameter"; usage; exit 1; }
 
 declare api1="${1}"
 declare api2="${2}"
 declare api3="${3}"
 
-# $1 = IP
+# $1 = endpoint
 # $2 = Hopr command
+# $3 = OPTIONAL: positive assertion message
+# $4 = OPTIONAL: maximum wait time in seconds during which we busy try
+# afterwards we fail, defaults to 0
+# $4 = OPTIONAL: step time between retries in seconds, defaults to 1 second
+# $5 = OPTIONAL: end time for busy wait in nanoseconds since epoch, has higher
+# priority than wait time, defaults to 0
```

Comment:
Can you explain?

Reply:
This is a helper parameter to be able to recursively call the function.